libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkClientImpl.cs

```diff
@@ -0,0 +1,131 @@
+                    ConversationType = activityClone.Conversation.ConversationType,
+                    AadObjectId = activityClone.Conversation.AadObjectId,
+                    IsGroup = activityClone.Conversation.IsGroup,
+                    Properties = activityClone.Conversation.Properties,
+                    Role = activityClone.Conversation.Role,
+                    TenantId = activityClone.Conversation.TenantId,
+                }
+            };
+            activityClone.Conversation.Id = conversationId;
+            activityClone.ServiceUrl = serviceUrl.ToString();
+            activityClone.Recipient ??= new ChannelAccount();
+            activityClone.Recipient.Role = RoleTypes.Skill;
+
+            // Create the HTTP request from the cloned Activity and send it to the Skill.
+            using (var jsonContent = new StringContent(JsonConvert.SerializeObject(activityClone, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }), Encoding.UTF8, "application/json"))
+            {
+                using (var httpRequestMessage = new HttpRequestMessage())
+                {
+                    httpRequestMessage.Method = HttpMethod.Post;
+                    httpRequestMessage.RequestUri = toUrl;
+                    httpRequestMessage.Content = jsonContent;
+
+                    // Add the auth header to the HTTP request.
+                    await credentials.ProcessHttpRequestAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+
+                    using (var httpResponseMessage = await _httpClient.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false))
+                    {
+                        var content = httpResponseMessage.Content != null ? await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false) : null;
+
+                        // Capture the HTTP response, assuming JSON in the response body, otherwise make it NULL.
+                        return new InvokeResponse<T>
+                        {
+                            Status = (int)httpResponseMessage.StatusCode,
+                            Body = content?.Length > 0 ? GetBodyContent<T>(content) : default
+                        };
+                    }
+                }
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            _httpClient.Dispose();
+            base.Dispose(disposing);
+            _disposed = true;
+        }
+
+        private static T GetBodyContent<T>(string content)
+        {
+            try
+            {
+                return JsonConvert.DeserializeObject<T>(content);
+            }
+            catch (JsonException)
+            {
+                // This will only happen when the skill didn't return valid json in the content (e.g. when the status code is 500 or there's a bug in the skill)
```

Comment:
Here if the http status code in the caller is non-success, means that the skill is failing but we are not failing anywhere. The result is that the bot never gets a response. but acts as if everything looks good, and there is no errors in logs.  We should do something about this!

Reply:
I think we should fix it for R13. Part of this PR or another?